docs: add oracle initialization docs (#611)
medibloc/panacea-core: .gitbook/5-oracles/1-operate-oracle-nodes/2-oracle-intialization.md

# A primary RPC address for light client verification

light-client-primary-addr = "tcp://127.0.0.1:26657"

# Witness addresses (comma-separated) for light client verification

light-client-witness-addrs = "tcp://127.0.0.1:26657"

# Setting log information for light client

light-client-log-level = "error"

###############################################################################
###                         IPFS Configuration                           ###
###############################################################################

[ipfs]

ipfs-node-addr = "127.0.0.1:5001"

###############################################################################
###                         API Configuration                           ###
###############################################################################

[api]

listen-addr = "127.0.0.1:8080"
write-timeout = "60"
read-timeout = "15"
```

## Configuring Some Default Setting

#### Base Configuration

In `Base Configuration`, you need to configure a `oracle-mnemonic`, `oracle-acc-num`, and `oracle-acc-index`. Those
components should be corresponded to account that you registered in the Panacea.

#### Panacea Configuration

> In `Panacea Configuration`, you need to configure a chain ID of Panacea. If you want to join the oracle in the Panacea
> mainnet, the chain ID will be configured to `panacea-3`. Or if you want to join in the Panacea testnet, the chain ID
> will be `hygeia-8`.

Reviewer: ```suggestion
In `Panacea Configuration`, you need to configure a chain ID of Panacea. If you want to join the oracle to Panacea mainnet or testnet, please check the chain IDs in [mainnet-repo](https://github.com/medibloc/panacea-mainnet#intro) or [testnet-repo](https://github.com/medibloc/panacea-testnet#intro), respectively.
```

Author: Thanks for detail feedback! I changed in [16db821](https://github.com/medibloc/panacea-core/pull/611/commits/16db8210675ad25780b64c9c31056115eb5ec8c1).

The default `grpc-addr` and `rpc-addr` setting is based on localnet. So if you want to connect with the Panacea

> mainnet, the `grpc-addr` will be `https://grpc.gopanacea.org` and the `rpc-addr` will be `https://rpc.gopanacea.org`.

Reviewer 2: Would be better to recommend them to use the endpoints of their own node. Plus, you can mention that they can use the public endpoint provided by the Panacea team: https://github.com/medibloc/panacea-mainnet#public-endpoints

Author: Yes, It could be more better for the users.

# Initialization

#### Before run the oracle, you should initialize the oracle configuration.

## Preparations

Before initializing the oracle, you should install the oracle described
in [oracle-installation](./1-oracle-installation.md).
After install the oracle, create an empty directory on your host, to be mounted as the `/home_mnt` directory in the
enclave.

```bash
mkdir <directory-you-want>/oracle
```

After that, if you want to run the `oracled` with Docker as described in
the [oracle-installation](./1-oracle-installation.md),
it is recommended to create an environment variable that you can execute the Docker container easily.

```bash
export ORACLE_CMD="docker run --rm \
  --device /dev/sgx_enclave \
  --device /dev/sgx_provision \

>   -v $(pwd)/oracle:/oracle ghcr.io/medibloc/panacea-oracle:latest \

Reviewer: ```suggestion
  -v <directory-you-want>/oracle:/oracle ghcr.io/medibloc/panacea-oracle:latest \
```

  ego run /usr/bin/oracled"
```

## Command Line of Initialization

```bash

> $DOCKER_CMD init --home $HOME/.oracle 

Reviewer: We don't have no longer `DOCKER_CMD` :)

> $DOCKER_CMD init --home $HOME/.oracle 

Reviewer: And the path for home probably should be `/home_mnt/.oracle`, right?.

Author: Right, I missed it.

# Initialization

#### Before run the oracle, you should initialize the oracle configuration.

## Preparations

Before initializing the oracle, you should install the oracle described
in [oracle-installation](./1-oracle-installation.md).
After install the oracle, create an empty directory on your host, to be mounted as the `/home_mnt` directory in the
enclave.

```bash
# If you run oracle on your host
sudo mkdir /oracle

> 

Reviewer 3: I think we could just delete the examples that don't use docker. 
Users will be confused.
wdyt?

Author: Applied in [2f0e3b3](https://github.com/medibloc/panacea-core/pull/611/commits/2f0e3b315aac2c1cebfc7a7cbda8ba0faea59a51).

# If you run oracle using Docker
mkdir $(pwd)/oracle
```

After that, if you want to run the `oracled` with Docker as described in
the [oracle-installation](./1-oracle-installation.md),
it is recommended to create an environment variable that you can execute the Docker container easily.

```bash
export DOCKER_CMD="docker run --rm \
  --device /dev/sgx_enclave \
  --device /dev/sgx_provision \

>   -v $(pwd)/oracle:/oracle ghcr.io/medibloc/panacea-oracle:latest"

Reviewer 3: If you change the cmd like @H4NLee said, it would be good to change it to `ORACLE_CMD` instead of `DOCKER_CMD`.

```

If you are not going to run the `oracled` without Docker, you can set an `DOCKER_CMD` environment variable empty string.

```bash

> export DOCKER_CMD=""

Reviewer: ```suggestion
export DOCKER_CMD="ego run oracled"
```

Author: I applied the https://github.com/medibloc/panacea-core/pull/611#discussion_r1066637127 comment, so I deleted this part.

```

## Command Line of Initialization

```bash
$DOCKER_CMD ego run oracled init --home $HOME/.oracle 
```

When run the above CLI for initializing the oracle, the `config.toml` file will be generated under the `$HOME/.oracle`
in the enclave.

> The `config.toml` file will be shown like this:

Reviewer: ```suggestion
The default `config.toml` file will be shown like this:
```

###############################################################################

[panacea]

chain-id = ""
grpc-addr = "http://127.0.0.1:9090"
rpc-addr = "tcp://127.0.0.1:26657"
default-gas-limit = "400000"
default-fee-amount = "2000000umed"

# A primary RPC address for light client verification

light-client-primary-addr = "tcp://127.0.0.1:26657"

# Witness addresses (comma-separated) for light client verification

light-client-witness-addrs = "tcp://127.0.0.1:26657"

# Setting log information for light client

light-client-log-level = "error"

###############################################################################
###                         IPFS Configuration                           ###
###############################################################################

[ipfs]

ipfs-node-addr = "127.0.0.1:5001"

###############################################################################
###                         API Configuration                           ###
###############################################################################

[api]

listen-addr = "127.0.0.1:8080"
write-timeout = "60"
read-timeout = "15"
```

> 

Reviewer: I think it would be nice if you could provide instructions on how to modify which field in `config.toml`, e.g. chain-id = "panacea-3".

Reviewer: The operator of oracle have to change some configurations, for example mnemonic, chain-id, etc.
So, I meant how about providing explanation for those configurations (must to be modified), excepting those with default values.

Author: Ah, does it mean that the argument is required or not?

Author: As we discussed offline, the every component of `config.toml` will be not introduced in this document. As similar as [cosmos sdk docs](https://docs.cosmos.network/v0.46/run-node/run-node.html), the only component that can be modified or updated will be introduced. The other components will be explained using comment in `config.toml`.

Author: I reflected in [3a2beb8](https://github.com/medibloc/panacea-core/pull/611/commits/3a2beb86691e3eaaf09fe07bc4fe9b961716354c). Please review again @H4NLee .
